compiler/src/iree/compiler/GlobalOptimization/DemoteContractionInputsToBF16.cpp

```diff
@@ -133,12 +135,13 @@ struct DemoteContractionInputsToBF16Pattern
 };
 
 class DemoteContractionInputsToBF16Pass
-    : public DemoteContractionInputsToBF16Base<
+    : public impl::DemoteContractionInputsToBF16PassBase<
           DemoteContractionInputsToBF16Pass> {
-
 public:
+  using impl::DemoteContractionInputsToBF16PassBase<
+      DemoteContractionInputsToBF16Pass>::DemoteContractionInputsToBF16PassBase;
   explicit DemoteContractionInputsToBF16Pass(const DemotionOption &option) {
-    this->demoteOnly.setValue(option);
+    this->demoteOnly = option;
   }
```

Comment:
Do we still need this extra constructor?

Reply:
We can use the tablegen option, but I found that it is a visual noise for some cases. Thus I created few wrapper and add `explicit` keyword to the constructor. My rule of thumb is: if the number of options is always one, it looks better to me; if we could potentially add more options to the pass, we prefer using tablegen option. What do you think?

Reply:
This is also why I ended up with few wrappers for CPU passes, e.g.,

https://github.com/iree-org/iree/blob/fbf677d2096aea4c544cd8d0558b49bfc729fd91/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.h#L22-L49

Reply:
I see, makes sense. I do agree that Passes.cpp can get cluttered with some of these options. Let's keep it like this then.

Reply:
(I opened the conversation because someone could have the same question and trace back to the PR. In this case, they will notice the conversation.)